controllers/controller.js

```diff
@@ -34,11 +34,28 @@ exports.getArticleById = (request, response, next) => {
             response.send({ article: rows[0] })
         })
         .catch((error) => {
-            if (error.status === 404) {
-                error.message = '404: article not found';
-                next(error);
-            } else {
-                next(error);
-            }
-        })
-};
\ No newline at end of file
+            next(error);
+        })
+};
+
+exports.patchArticleById = (request, response, next) => {
+    const { params: { article_id }, body } = request
+    const entries = Object.entries(body);
+
+    checkParametricFormat(article_id, 'article_id', 'articles')
+        .then(() => {
+            return checkParametricValue(article_id, 'article_id', 'articles')
```

Comment:
I wanted to, to speed it up and streamline the code, but I ran into an issue. My custom error handler relies on each promise resolving sequentially, otherwise my error message responses could be misleading.

For example, if I promise all, the first checker to come back (format or value) will throw the error and determine my final error message.

An improperly formatted parameter would foul the value testing function, and if that resolved first, would response with a 404, rather than a 400.